packages/vite/src/node/plugins/terser.ts

```diff
@@ -2,18 +2,29 @@ import { Worker } from 'okie'
 import type { Terser } from 'types/terser'
 import type { Plugin } from '../plugin'
 import type { ResolvedConfig } from '..'
+import { requireResolveFromRootWithFallback } from '../utils'
+
+let terserPath: string | undefined
+const loadTerserPath = (root: string) => {
+  if (terserPath) return terserPath
+  try {
+    terserPath = requireResolveFromRootWithFallback(root, 'terser')
+  } catch (e) {
+    if (e.code === 'MODULE_NOT_FOUND') {
+      throw new Error('terser not found. Did you install it? Since v3.0.0, terser has become a optional dependency and needs to be installed individually.')
```

Comment:
Maybe we could simplify the wording a bit?
```suggestion
      throw new Error('terser not found. Since Vite v3, terser has become an optional dependency. You need to install it.')
```